### PR TITLE
TINKERPOP-1866 Support g:T for .NET

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-1866

Issuing this PR for @FlorianHockmann now that TINKERPOP-1865 has merged. 

Builds with ` mvn clean install -DskipTests && mvn verify -pl :gremlin-dotnet-tests -DskipIntegrationTests=false`

VOTE +1

### CODE REVIEW

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/CommonSteps.cs [hunk 1]
```diff
@@ -239,7 +240,8 @@ private static object ToLambda(string stringLambda, string graphName)
 
         private static object ToT(string enumName, string graphName)
```
Comment (dkuppitz) [inline gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/CommonSteps.cs]: What's the point of the `graphName` argument?

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/ScenarioData.cs [hunk 2]
```diff
@@ -30,6 +30,7 @@
 using Gremlin.Net.Process.Remote;
 using Gremlin.Net.Process.Traversal;
 using Gremlin.Net.Structure;
+using static Gremlin.Net.Process.Traversal.__;
```
Comment (dkuppitz) [inline gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/ScenarioData.cs]: Seems like an accident, as it wasn't needed before and nothing else has changed in this file.

Comment (spmallette) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/ScenarioData.cs]: i saw that, but didn't kill it b/c it was Florian's changes. i guess i should have questioned it. i will try to remove that change locally and see what happens.

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/CommonSteps.cs (hunk 1 above)
Comment (spmallette) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/CommonSteps.cs]: It's not really useful to this conversion, but it's the expected signature for the lambda where this method is used. See further up in the definition of `Parsers` where it requires a `Func<string,string,object>` a two arg function basically. maybe we could make that nicer, but this is all plumbing code for tests, so perhaps not a big issue?

Comment (dkuppitz) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/CommonSteps.cs]: No, not an issue, I was just curious.

**Comment by dkuppitz** [comment]
VOTE: +1

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/ScenarioData.cs (hunk 2 above)
Comment (spmallette) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/ScenarioData.cs]: that commit was irrelevant - killed it out and force pushed

**Pull request merged**